lib/internal/url.js

```diff
@@ -1406,14 +1408,26 @@ function getPathFromURL(path) {
 // as this is the only character that won't be percent encoded by
 // default URL percent encoding when pathname is set.
 const percentRegEx = /%/g;
-function getURLFromFilePath(filepath) {
+function pathToFileURL(filepath) {
+  let resolved = path.resolve(filepath);
+  // path.resolve strips trailing slashes so we must add them back
+  if (filepath[filepath.length - 1] === '/' &&
```

Comment:
☝️ The checks above for slashes are usually done with char codes instead of string values.

Reply:
I can certainly switch to codes, sure.